ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1199,19 +1209,19 @@ func (r *RayClusterReconciler) calculateStatus(ctx context.Context, instance *ra
 	newInstance.Status.DesiredGPU = sumGPUs(totalResources)
 	newInstance.Status.DesiredTPU = totalResources[corev1.ResourceName("google.com/tpu")]
 
-	if utils.CheckAllPodsRunning(ctx, runtimePods) {
+	if utils.CheckAllPodsRunning(ctx, corev1.PodList{Items: append(headPods.Items, workerPods.Items...)}) {
```

Comment:
The best way is to use headPod and workerPods separately to decide whether RayCluster is in 'ready' status. However, I decided to keep the original logic as much as I can to avoid any potential breaking changes, and we will soon remove or refactor this status anyway.